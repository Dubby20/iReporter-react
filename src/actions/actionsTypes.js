/* eslint-disable no-unused-expressions */
export const REGISTER_SUCCESS = 'REGISTER_SUCCESS';
export const REGISTER_BEGIN = 'REGISTER_BEGIN';
export const REGISTER_DONE = 'REGISTER_DONE';
export const NEW_NOTIFICATION = 'NEW_NOTIFICATION';
export const CLEAR_NOTIFICATION = 'CLEAR_NOTIFICATION';
export const LOGIN_SUCCESS = 'REGISTER_SUCCESS';
export const LOGIN_BEGIN = 'REGISTER_BEGIN';
export const LOGIN_DONE = 'REGISTER_DONE';
export const GET_REDFLAG = 'GET_REDFLAG';
export const GET_INTERVENTION = 'GET_INTERVENTION';
export const GET_SINGLE_RECORD = 'GET_SINGLE_RECORD';
export const START_FETCHING = "START_FETCHING";
export const STOP_FETCHING = "STOP_FETCHING";
export const EDIT_COMMENT = "EDIT_COMMENT";
export const ADMIN_RECORDS = "ADMIN_RECORDS";
export const PROFILE_HISTORY = "PROFILE_HISTORY";
export const GET_LOCATION = 'GET_LOCATION';
export const POST_REPORT = 'POST_REPORT';
export const LOG_OUT = "LOG_OUT";
