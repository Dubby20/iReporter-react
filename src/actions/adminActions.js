import axios from 'axios';
import {
  ADMIN_RECORDS
} from './actionsTypes';
import apiUrl from '../services/index';


const adminRequest = () => async (dispatch, getState) => {
  const urls = [
    `${apiUrl}/interventions`,
    `${apiUrl}/red-flags`
  ];

  try {
    const { token } = getState().authReducer.user;
    const promises = await Promise.all(urls.map(url => axios.get(`${url}`, {
      headers: { 'x-access-token': token }
    })));

    const intervention = promises[0].data.data[0].intervention;
    const redFlag = promises[1].data.data[0].redFlag;

    const merge = [...intervention, ...redFlag];
    const allIncident = merge.sort((a, b) => a.id - b.id);


    dispatch({
      type: ADMIN_RECORDS,
      records: allIncident,
    });

    return promises;
  } catch (error) {
    return error.response;
  }
};

export default adminRequest;
