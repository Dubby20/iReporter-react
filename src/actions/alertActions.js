const SUCCESS = 'SUCCESS';
const FAILURE = 'FAILURE';

// export const success = message => ({
//   type: SUCCESS,
//   message
// });


// export const failure = message => ({
//   type: FAILURE,
//   message
// });
