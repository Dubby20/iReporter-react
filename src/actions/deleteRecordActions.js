import axios from 'axios';
import apiUrl from '../services/index';


const deleteRecordRequest = (id) => async (dispatch, getState) => {
  let recordType = localStorage.getItem('recordType');
  let url;

  if (recordType === 'red-flag') {
    url = `${apiUrl}/red-flags/${id}/`;

    recordType = 'Red-Flag';
  } else if (recordType === 'intervention') {
    url = `${apiUrl}/interventions/${id}/`;
    recordType = 'Intervention';
  }

  try {

    const { token } = getState().authReducer.user;
    const { data } = await axios.delete(`${url}`, {
      headers: { 'x-access-token': token }
    });


    return data;
  } catch (error) {
    return error.response;
  }
};


export default deleteRecordRequest;
