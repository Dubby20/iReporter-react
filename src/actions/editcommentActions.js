import axios from 'axios';
import {
  GET_SINGLE_RECORD,
} from './actionsTypes';
import apiUrl from '../services/index';
import { clearNotification, newNotification } from '../services/notificationServices';


const editCommentRequest = ({ id, type, newComment }) => async (dispatch, getState) => {
  let url;
  if (type === 'red-flag') {
    url = `${apiUrl}/red-flags/${id}/comment`;

  } else if (type === 'intervention') {
    url = `${apiUrl}/interventions/${id}/comment`;
  }

  try {
    dispatch(clearNotification());

    const { token } = getState().authReducer.user;
    const { data } = await axios.patch(`${url}`, { comment: newComment }, {
      headers: { 'x-access-token': token }
    });

    const recordDetails = data.data[0].editComment;

    dispatch({
      type: GET_SINGLE_RECORD,
      record: recordDetails,
    });

    return data;
  } catch (error) {
    dispatch(newNotification({
      message: error.response.data.error,
      notificationType: 'error',
    }));
  }
};

export default editCommentRequest;
