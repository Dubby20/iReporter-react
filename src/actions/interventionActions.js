import axios from 'axios';
import {
  GET_INTERVENTION,
  START_FETCHING,
  STOP_FETCHING,
} from './actionsTypes';
import apiUrl from '../services/index';



export const interventionRequest = () => async (dispatch, getState) => {

  try {
    dispatch({ type: START_FETCHING });
    const { token } = getState().authReducer.user;
    const { data } = await axios.get(`${apiUrl}/interventions`, {
      headers: { 'x-access-token': token }
    });
    localStorage.setItem('recordType', data.data[0].intervention[0].type);
    const recordList = data.data[0].intervention;

    dispatch({
      type: GET_INTERVENTION,
      records: recordList,
    });

    dispatch({ type: STOP_FETCHING });
    return data;
  } catch (error) {
    return error.response;
  }
};

