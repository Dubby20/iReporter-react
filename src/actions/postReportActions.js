/* eslint-disable import/prefer-default-export */
import axios from 'axios';
import {
  START_FETCHING,
  STOP_FETCHING,
  POST_REPORT
} from './actionsTypes';
import { clearNotification, newNotification } from '../services/notificationServices';
import { uploadFile } from '../helpers/utils';
import apiUrl from '../services/index';



export const postReport = ({ comment, reportType, image, video, location }) => async (dispatch, getState) => {

  let url;
  if (reportType === 'red-flag') {
    url = `${apiUrl}/red-flags`;

  } else if (reportType === 'intervention') {
    url = `${apiUrl}/interventions`;
  }

  try {
    const images = [];
    const videos = [];

    dispatch(clearNotification());
    dispatch({ type: START_FETCHING });

    if (image.length > 0) {
      const imageResponse = await uploadFile(image, 'https://api.cloudinary.com/v1_1/djdsxql5q/image/upload');
      images.push(imageResponse);
    }
    if (video.length > 0) {
      const videoResponse = await uploadFile(video, 'https://api.cloudinary.com/v1_1/djdsxql5q/video/upload');
      videos.push(videoResponse);
    }

    const { token } = getState().authReducer.user;
    const { data } = await axios.post(`${url}`, { comment, reportType, images, videos, location }, {
      headers: { 'x-access-token': token }
    });

    dispatch({
      type: POST_REPORT,
      report: data,
    });

    dispatch({ type: STOP_FETCHING });
    return data;

  } catch (error) {
    dispatch(newNotification({
      message: error.response.data.error,
      notificationType: 'error',
    }));
  }
};

