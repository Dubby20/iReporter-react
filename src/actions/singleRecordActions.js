import axios from 'axios';
import {
  GET_SINGLE_RECORD,
  START_FETCHING,
  STOP_FETCHING,
} from './actionsTypes';
import apiUrl from '../services/index';


export const setRecord = record => ({
  type: GET_SINGLE_RECORD,
  record,
});

export const singleRecordRequest = ({ id, type }) => async (dispatch, getState) => {
  let recordUrl;

  if (type === 'red-flag') {
    recordUrl = `${apiUrl}/red-flags/${id}`;
  } else if (type === 'intervention') {
    recordUrl = `${apiUrl}/interventions/${id}`;
  }
  try {
    dispatch({ type: START_FETCHING });

    const { token } = getState().authReducer.user;
    const { data } = await axios.get(`${recordUrl}`, {
      headers: { 'x-access-token': token }
    });
    const recordList = data.data[0].report;

    dispatch({
      type: GET_SINGLE_RECORD,
      record: recordList,
    });

    dispatch({ type: STOP_FETCHING });
    return data;
  } catch (error) {
    return error.response;
  }
};
