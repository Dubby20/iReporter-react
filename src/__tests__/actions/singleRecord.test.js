import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import mockAxios from 'axios';
import * as types from '../../actions/actionsTypes';
import * as actions from '../../actions/singleRecordActions';


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('handles singleRecordRequest', () => {
  it('handles success case', () => {
    const mockData = {
      report: {
        type: 'intervention',
        comment: "This is a comment",
        images: ["https://imgur.com"],
        videos: ["res.cloudinary"],
        location: "6.7868, 7.8993",
        status: 'draft',
      }
    };

    mockAxios.get.mockImplementationOnce(() => Promise.resolve({
      data: mockData
    }));

    const expectedActions = {
      type: types.START_FETCHING,
      report: mockData.recordList
    };

    const store = mockStore({});

    return store.dispatch(actions.singleRecordRequest("id", "type")).then(() => {
      expect(store.getActions()).toEqual([expectedActions]);
    });
  });


});
