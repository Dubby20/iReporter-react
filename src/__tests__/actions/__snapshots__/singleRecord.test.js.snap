// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles singleRecordRequest handles success case 1`] = `
Array [
  Object {
    "type": "START_FETCHING",
  },
]
`;
